Source Files/second_pass.c

```diff
@@ -0,0 +1,41 @@
+#include "../Header Files/second_pass.h"
+
+int secondPass(FILE *inputFile, char *inputFileName, LinkedList *symbolTable, int *binaryCodesTable)
+{
+    int IC = 0;
+    char line[MAX_LINE_LEN];
+    AssemblyLine parsedLine;
+
+    /* TODO: maybe dont go over all the lines, and just go over the symbolTable + binaryCodesTable */
```

Comment:
Ok, i think i get it.
First of all we can create a list of parsed lines and pass it from the first pass to the second.
Secondly, in my opinion the algorithm should be like this:
1. first pass - create a list of all the parsed lines + symbol table
2. second pass:
go over all of the parsed lines and create a binary representation according to the adressing options and symbol names.
For example, i get the line START: mov r3,SIZE[5]. it is already parsed so i can ignore the label altogether.
i ignore the START and create one binary word for mov, one for r3 and two for SIZE[5]. notice that i will have to check if SIZE exists in the Symbol table before this line (i think) and if not raise an error.
then to the next one:
data 1,2,6,12 - ignore
 prn #-5 - one word for prn, one for #-5
this is just an example not realy an algorithm but i can pretty much see it in my head. just go over every parsed line and create a binary words for her and check for exceptions in the order of the labels sor some shit.